geozero/src/wkb/wkb_reader.rs

```diff
@@ -70,29 +106,28 @@ pub fn process_wkb_type_geom<R: Read, P: GeomProcessor>(
         WkbDialect::Wkb => process_wkb_geom(raw, processor),
         WkbDialect::Ewkb => process_ewkb_geom(raw, processor),
         WkbDialect::Geopackage => process_gpkg_geom(raw, processor),
+        WkbDialect::Spatialite => process_spatialite_geom(raw, processor),
+        WkbDialect::MySQL => process_mysql_geom(raw, processor),
     }
 }
 
 #[derive(Debug)]
 pub(crate) struct WkbInfo {
-    endian: scroll::Endian,
+    endian: Endian,
     base_type: WKBGeometryType,
     has_z: bool,
     has_m: bool,
     #[allow(dead_code)]
     srid: Option<i32>,
     #[allow(dead_code)]
     envelope: Vec<f64>,
+    is_compressed: bool,
 }
 
 /// OGC WKB header.
 pub(crate) fn read_wkb_header<R: Read>(raw: &mut R) -> Result<WkbInfo> {
     let byte_order = raw.ioread::<u8>()?;
-    let endian = if byte_order == WKBByteOrder::Xdr as u8 {
-        scroll::BE
-    } else {
-        scroll::LE
-    };
+    let endian = Endian::from(byte_order != 0);
```

Comment:
sigh, `Endian::from(bool)` is not the most readable interface. I had to look up what `true` means :(

Reply:
P.S. this one is not exactly actionable - just a note in case you can think of a cleaner API usage here

Reply:
Yup, I agree that `Endian::from(bool)` is not the very explicit at first glance... If we implement the reverse of `From<scroll::Endian> for WKBByteOrder` as you suggested, then maybe we could also implement `From<u8> for WKBByteOrder` and chain the two. That's a lot of code for basically a bool check though so...

Reply:
yeah, probably not worth the complexity - maybe just add an extra line to make it more readable, like this?

```rust
let is_little_endian = byte_order != 0;
let endian = Endian::from(is_little_endian);
```

Reply:
Much better, I'll do that 👍